src/main/java/edu/harvard/iq/dataverse/util/DataSourceProducer.java

```diff
@@ -0,0 +1,31 @@
+package edu.harvard.iq.dataverse.util;
+
+import javax.annotation.Resource;
+import javax.annotation.sql.DataSourceDefinition;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+import javax.sql.DataSource;
+
+@Singleton
+@DataSourceDefinition(
+    name = "java:global/jdbc/dataverse",
+    // Using PGXADataSource instead of deprecated PGPoolingDataSource
+    className = "org.postgresql.xa.PGXADataSource",
+    user = "${MPCONFIG=dataverse.db.user}",
+    // Has to be using ALIAS for at least Payara 5.2020.3, as MPCONFIG does not seem to read properly
```

Comment:
This has been solved as of Payara 5.2020.4.
We should use the fixed version of Payara and implement this in a clean way instead of relying on the workaround.
MicroProfile Config support needs to be adressed ASAP IMHO.

Reply:
FWIW integration tests on 5.2020.5 produce the same results as 5.2020.2:

<img width="455" alt="Screen Shot 2020-10-23 at 10 43 06" src="https://user-images.githubusercontent.com/12831666/97018543-432a1b80-151d-11eb-8171-2e7aded319af.png">

GitHub won't let me attach the HTML report, even zipped, but I can e-mail or Slack it if anyone likes.

Reply:
Could you please clarify the "solved as of Payara 5.2020.4" part? Does that mean the PGXADataSource is just not supported in earlier versions?

Reply:
OK,  got it,  the alias part.

Reply:
@landreev this is just Github highlighting stuff that is in the context. The comment is about L15-17, see https://github.com/poikilotherm/dataverse/blob/2e6b6f9f4815e8c53fe59787b2b273aac3ddc24a/src/main/java/edu/harvard/iq/dataverse/util/DataSourceProducer.java#L15-L17

Reply:
Unless I'm missing something, it looks like this PR leaves "asadmin ... create-jdbc-connection-pool" command in the installer? Should it be removed?

Reply:
> @landreev you mentioned `validation-table-name="setting" is-connection-validation-required="true"` above.
> Should we introduce those to `persistance.xml` and make it configurable?
> 
> See also https://github.com/payara/Payara-Server-Documentation/blob/master/documentation/payara-server/advanced-jdbc/advanced-connection-pool-properties.adoc

I'm not sure about this part. It looks like we never felt these validation settings were important enough, as to recommend them to everybody... So maybe we shouldn't eve worry about it too much.

Reply:
And what about any existing installations - do they need to explicitly remove their existing `<jdbc-connection-pool ... ` sections from their configuration? Or is it ok to leave them in place?

Reply:
It would be ok to leave them on place. I added some docs how to remove them anyway.